intrahospital_api/management/commands/merge_all_patients.py

```diff
@@ -0,0 +1,166 @@
+from django.db import transaction
+from opal.models import Patient
+from intrahospital_api.apis.prod_api import ProdApi as ProdAPI
+from intrahospital_api import logger, loader, merge_patient, update_demographics
+from elcid import episode_categories as elcid_episode_categories
+from elcid.utils import timing
+from elcid import models
+
+
+# Returns all active merged patients
+# used by the merge_all_patients mgmt
+# command
+GET_ALL_ACTIVE_MERGED_MRNS = """
+    SELECT Patient_Number FROM CRS_Patient_Masterfile
+    WHERE MERGED = 'Y'
+    AND ACTIVE_INACTIVE = 'ACTIVE'
+"""
+
+def get_all_active_merged_mrns():
+    api = ProdAPI()
+    query_result = api.execute_hospital_query(GET_ALL_ACTIVE_MERGED_MRNS)
+    return [i["Patient_Number"] for i in query_result]
+
+
+
+@transaction.atomic
+def our_merge_patient(*, old_patient, new_patient):
+    """
+    All elcid native non-singleton entries to be converted to
+    the equivalent episode category on the wining MRN, with a reference
+    created in the original_mrn field where they have been moved
+
+    Copy teams from the old infection service episode? To the new
+    Infection service episode
+
+    Copy over any episode categories that do not exist, iterate
+    over subrecord attached to these and add the PreviousMRN
+
+    Singleton entries to pick the latest but create a reversion
+    history entry for the non-oldest, with a reference to the original_mrn
+
+    Non-singletons entries are moved from the old parent to the
+    new parent.
+
+    This is the same as merge_patient.merge_patient but
+    does not call load_patient.
+    """
+    old_mrn = old_patient.demographics().hospital_number
+    for patient_related_model in merge_patient.PATIENT_RELATED_MODELS:
+        merge_patient.move_record(
+            patient_related_model,
+            old_patient,
+            new_patient,
+            old_mrn,
+        )
+    for old_episode in old_patient.episode_set.all():
+        # Note: if the old episode has multiple episode
+        # categories of the same category name
+        # this will merge those.
+        new_episode, _ = new_patient.episode_set.get_or_create(
+            category_name=old_episode.category_name
+        )
+        merge_patient.update_tagging(old_episode, new_episode)
+        for episode_related_model in merge_patient.EPISODE_RELATED_MODELS:
+            merge_patient.move_record(
+                episode_related_model,
+                old_episode,
+                new_episode,
+                old_mrn,
+            )
+    old_patient.delete()
+    new_patient.mergedmrn_set.filter(mrn=old_mrn).update(
+        our_merge_datetime=timezone.now()
+    )
+    merge_patient.updates_statuses(new_patient)
+
+
+@transaction.atomic
+def check_and_handle_upstream_merges_for_mrns(mrns):
+    """
+    Takes in a list of MRNs.
+
+    Filters those not related to elCID.
+
+    If they are inactive, creates a Patient for the
```

Comment:
Is this accurate to the implementation isn't it filtered on active in the query?